### ISSUE TITLE
documentation for compilation and install

### ISSUE THREAD

**Issue by tsequitur** [rpclib/rpclib#173]
hi, i am trying to install rpclib, but got a bit lost.

the version i am trying to install is the git cloned version from the master branch.
i follow the [guide](https://github.com/rpclib/rpclib/blob/master/doc/pages/compiling.md) and execute the build commands. the documentation says
> And that's it. If all goes well, your build output will be in the `output` directory.

i can find no directory named output.
the documentation also describes no procedure for installation or which files would need to be installed.
looking in the build directory, i guess i need to copy the .a file (i've only dealt with .so files in the past) and move the quite big (7.3MB) ../include/rpc directory to /usr/include, is that correct?

another thing is that the [getting started](http://rpclib.net/gettingstarted/) page says
> There are various release packages available for the most popular systems and ideally you should install one of those.

i use arch linux and there does not seem to be a package available. out of curiosity i also searched the internet for "rpclib package" "ubuntu rpclib" "fedora rpclib" "debian rpclib" and find no reference to any package. just now i find some lonely .deb files under releases, but only for v1.0.0 preview. which makes me wonder if there actually are packages for the most popular systems.

**Comment by sztomi** [comment]
Thanks for bringing that to my attention. That was changed recently and the documentation was not updated. I think the easiest way is to specify `cmake -DCMAKE_INSTALL_PREFIX=<whatever directory you want the output in>` then run `make install` (or the equivalent thereof for the particular generator you used).

**Comment by sztomi** [comment]
Also,

> i guess i need to copy the .a file (i've only dealt with .so files in the past) and move the quite big (7.3MB) ../include/rpc directory to for example /usr/include, is that correct?

It's best not to do that, and instead add the installation directory/include to your include path, as well as the lib directory to your library search paths.

**Comment by tsequitur** [comment]
thanks for the response, i think it works now.

> cmake -DCMAKE_INSTALL_PREFIX=<whatever directory you want the output in>

this works except that it writes the full installation path into `lib/pkgconfig/rpclib.pc`, which is undesirable when installing to a temporary location for example when the result is to be included in a package archive. but it seems i can fix that with sed. i am also creating an arch linux package while i figure out the setup process which i submit to the arch user repository so users can install rpclib as a package with simple commands ([aur page](https://aur.archlinux.org/packages/rpclib-git/), [package source](https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=rpclib-git)).

i see it installs .cmake files like /usr/lib/cmake/rpclib/rpclibTargets-noconfig.cmake, not sure if that is intended.

> It's best not to do that, and instead add the installation directory/include to your include path, as well as the lib directory to your library search paths.

that's good advise, much easier to uninstall.

**Comment by MayaPosch** [comment]
Is it correct to assume that there is no Ubuntu package for rpclib?

**Comment by sztomi** [comment]
@MayaPosch yes. Although it is not hard to make one with CPack. But, in my opinion it is not very useful, unless it is included in Debian (but that wouldn't happen as it is due to the bundled deps).
